### PR TITLE
feat: wip on noarch and runtime deps

### PR DESCRIPTION
**Pull request by nichmor**
Checklist
* [ ] Added a ``news`` entry
* [ ] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

### CODE REVIEW

**Review by beckermr — CHANGES_REQUESTED** [review]
We do allow platform-only selectors in noarch recipes. Our current linter should have that edge case as a guide.

**Comment by nichmor** [comment]
> We do allow platform-only selectors in noarch recipes. Our current linter should have that edge case as a guide.

added lint for noarch platforms use-case. could you please take a look?

**Comment by wolfv** [comment]
Fixed up the merge conflicts and rebased on main.

File: news/2001-lint-noarch-deps.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+**Added:**
+
+* add lint for noarch and runtime deps (#2001)
```
Comment (beckermr) [inline news/2001-lint-noarch-deps.rst:3]: Let's add this the single item we already have going.

**Review by beckermr — CHANGES_REQUESTED** [review]
One nit.

**Pull request merged**